timesketch/tsctl.py

```diff
@@ -616,3 +622,91 @@ def validate_context_links_conf(path):
             print(f'=> OK: "{entry}"')
         except (ValidationError, SchemaError) as err:
             print(f'=> ERROR: "{entry}" >> {err}\n')
+
+
+# Analyzer stats cli command
+@cli.command(name="analyzer-stats")
+@click.argument("analyzer_name")
+@click.option(
+    "--timeline_id",
+    required=False,
+    help="Timeline ID if the analyzer results should be filtered by timeline.",
+)
+@click.option(
+    "--scope",
+    required=False,
+    help="Scope on: [many_hits, long_runtime, recent]",
+)
+@click.option(
+    "--result_text_search",
+    required=False,
+    help="Search in result text. E.g. for a specific rule_id.",
+)
+@click.option(
+    "--limit",
+    required=False,
+    help="Limit the number of results.",
+)
+def sigma_stats(analyzer_name, timeline_id, scope, result_text_search, limit):
+    """Prints sigma analyzer stats."""
+
+    if timeline_id:
+        timeline = Timeline.query.get(timeline_id)
+        if not timeline:
+            print("No timeline found with this ID.")
+            return
+        analysis_history = Analysis.query.filter_by(
+            timeline=timeline, analyzer_name=analyzer_name
```

Comment:
If I run the command with the parameters `--timeline_id X all` I would expect it to return all analysis that was run on this timeline. But with this query it will try to find an analysis with the name "all" on the timeline, what is a different behaviour than expected with "all" in the check below.
I would recommend to add another check for all in here and adjust the query accordingly.

Reply:
true, I missed that aspect, fixing it